### ISSUE TITLE
Some schemas cause infinite recursion in Mantle 2.0.3

### ISSUE THREAD

**Issue by cspickert** [Mantle/Mantle#581]
Something changed between Mantle 2.0.2 and 2.0.3 that is causing infinite recursion when parsing a recursive schema. I've created [a sample project](https://github.com/cspickert/mantle-bug) that reproduces the problem, but here are the essentials:

Model:

```objc
@interface User : MTLModel

@property (nonatomic, readonly) NSString *name;
@property (nonatomic, readonly) NSArray *groups; // optional array of Group objects

@end

@interface Group : MTLModel

@property (nonatomic, readonly) User *owner;
@property (nonatomic, readonly) NSArray *users;

@end
```

Data example 1 (`User`s contained in a `Group`):

```json
{
    "owner": {"name": "Cameron"},
    "users": [
        {"name": "Dimitri"}, {"name": "John"}
    ]
}
```

Data example 2 (`Group`s contained in a `User`):

```json
{
    "name": "Cameron",
    "groups": [
        {
            "users": [
                {"name": "Dimitri"}, {"name": "John"}
            ]
        }
    ]
}
```

I realize I could work around this problem by breaking the cycle in my schema, but I would prefer to use the same classes to represent both top-level as well as nested objects.

**Comment by jmah** [comment]
Thanks for the example, I'll check this out over the next couple of hours.

**Comment by robb** [comment]
I believe this to be fixed in `2.0.4`. Please reopen the issue if the problem persists.
